pkg/utils/tasks/tasks.go

```diff
@@ -34,6 +40,38 @@ type TaskManager struct {
 	stopped bool
 }
 
+// Option allows retries and backoff to be configured for individual tasks.
+type Option func(*options)
+
+type options struct {
+	Retries int
+	Backoff backoff.BackOff
+	ctx     *gin.Context
+	err     error
+}
+
+// Number of retries to attempt before giving up, default 0
+func WithRetries(retries int) Option {
+	return func(o *options) {
+		o.Retries = retries
+	}
+}
+
+// Backoff strategy to use when retrying, default is no backoff
+func WithBackoff(backoff backoff.BackOff) Option {
+	return func(o *options) {
+		o.Backoff = backoff
+	}
+}
```

Comment:
Can we make the default exponential backoff -- this is the one we're going to use most I think. If we need no backoff in the future; we can make a struct that implements `backoff.Backoff` that has no timeout (if it doesn't already exist in the cenkalti package, and it might).